### PR TITLE
Schema order fix

### PR DESCRIPTION
**Pull request by LtChae**
These changes create a field to track the order of fields in a schema. They are autogenerated as an increasing integer whenever the schema is edited.

The sequence field itself is editable, so we can easily use it to implement a reordering flow on the ui in the future if needed.

### CODE REVIEW

File: apps/andi/lib/andi/input_schemas/datasets/data_dictionary.ex [hunk 1]
```diff
@@ -21,6 +21,7 @@ defmodule Andi.InputSchemas.Datasets.DataDictionary do
     field(:rationale, :string)
     field(:bread_crumb, :string)
     field(:format, :string)
+    field(:sequence, :integer, read_after_writes: true)
```
Comment (bbrewer-forge) [inline apps/andi/lib/andi/input_schemas/datasets/data_dictionary.ex:24]: nice!

File: apps/andi/lib/andi/input_schemas/struct_tools.ex [hunk 2]
```diff
@@ -71,6 +71,12 @@ defmodule Andi.InputSchemas.StructTools do
     struct(struct_type, preloaded)
   end
 
+  defp sort_if_sequenced([%{sequence: _sequence} | _] = list) do
```
Comment (bbrewer-forge) [inline apps/andi/lib/andi/input_schemas/struct_tools.ex:74]: Cool, nicely done

File: apps/andi/lib/andi/input_schemas/struct_tools.ex [hunk 3]
```diff
@@ -71,6 +71,12 @@ defmodule Andi.InputSchemas.StructTools do
...
+    Enum.sort(list, fn %{sequence: sequence1}, %{sequence: sequence2} -> sequence1 < sequence2 end)
```
Comment (bbrewer-forge) [inline apps/andi/lib/andi/input_schemas/struct_tools.ex:75]: Is there a reason why `Enum.sort_by(&Map.get(&1, :sequence))` doesn't work here?

File: apps/andi/priv/repo/migrations/20200709175229_add_sequence.exs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+defmodule Andi.Repo.Migrations.AddSequence do
+  use Ecto.Migration
+
+  def change do
+    alter table(:data_dictionary) do
+      add :sequence, :serial
+    end
+    execute "select setval(pg_get_serial_sequence('data_dictionary', 'sequence'), 1)"
```
Comment (bbrewer-forge) [inline apps/andi/priv/repo/migrations/20200709175229_add_sequence.exs:8]: wowowow

File: apps/andi/lib/andi/input_schemas/struct_tools.ex (hunk 3 above)
Comment (LtChae) [reply apps/andi/lib/andi/input_schemas/struct_tools.ex:75]: Probably would. I'll look into it

**Pull request merged**